### PR TITLE
Feature/class structure

### PR DESCRIPTION
**Pull request by alejandroimperatori**

### CODE REVIEW

File: src/com/bomberman/entities/BaseObject.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	protected int x;
+	protected int y;
+	protected boolean destroyed;
+	protected boolean canBeDestroyed;
+	protected GameMap map;
+
+	public BaseObject(int x, int y, boolean destroyed, boolean canBeDestroyed, GameMap map) {
+		this.x = x;
+		this.y = y;
+		this.destroyed = destroyed;
```
Comment (matextrem) [inline src/com/bomberman/entities/BaseObject.java]: This.destroyed should be false as default.

**Comment by arieldamian** [comment]
Che y si al BaseObject le ponemos `Entity`? Me hace más sentido...

File: src/com/bomberman/entities/GameMap.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+	public void addObject(BaseObject obj) {
+		this.objects.add(obj);
+	}
+
+	public BaseObject getAtPosition(int x, int y) {
+		return this.objects.stream().filter(o -> o.x == x && o.y == y).findFirst().orElse(null);
+	}
+
+	public boolean isPlayerAlive() {
```
Comment (matextrem) [inline src/com/bomberman/entities/GameMap.java]: Should it receive a Player instance?

File: src/com/bomberman/entities/BaseObject.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.bomberman.entities;
+
+public abstract class BaseObject {
```
Comment (arieldamian) [inline src/com/bomberman/entities/BaseObject.java]: I propose to change `BaseObject` by `Entity`. What do you think? @alejandroimperatori @matextrem @marceloatorres

File: src/com/bomberman/entities/GameMap.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+package com.bomberman.entities;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GameMap {
```
Comment (arieldamian) [inline src/com/bomberman/entities/GameMap.java]: It's understood that it is from the "Game". Maybe we could use just `Map`...

File: src/com/bomberman/entities/GameMap.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	public GameMap(String name, int width, int height) {
+		this.name = name;
+		this.objects = new ArrayList<>();
+		this.width = width;
+		this.height = height;
+	}
+
+	public boolean canMove(int x, int y, Direction direction) {
+		switch (direction) {
+			case RIGHT:
```
Comment (matextrem) [inline src/com/bomberman/entities/GameMap.java]: It should return the inverse condition.

File: src/com/bomberman/entities/GameMap.java (hunk 2 above)
Comment (arieldamian) [reply src/com/bomberman/entities/GameMap.java]: This method is useless indeed. We should remove it. 

The Map doesn't need to know if a Player is alive or not.

File: src/com/bomberman/entities/BaseObject.java (hunk 3 above)
Comment (matextrem) [reply src/com/bomberman/entities/BaseObject.java]: It's okay for me. Entity is more significant name as well.

Comment (marceloatorres) [reply src/com/bomberman/entities/BaseObject.java]: I think Entity is more appropiate.

File: src/com/bomberman/entities/BaseObject.java (hunk 1 above)
Comment (alejandroimperatori) [reply src/com/bomberman/entities/BaseObject.java]: I agree, i will correct it

File: src/com/bomberman/entities/GameMap.java (hunk 4 above)
Comment (alejandroimperatori) [reply src/com/bomberman/entities/GameMap.java]: JAVA already has a class called Map, I don't think it would be best to name it that way

Comment (arieldamian) [reply src/com/bomberman/entities/GameMap.java]: Ooops, that's true! My bad!

**Comment by arieldamian** [comment]
It's ok for me. Guys? @marceloatorres @matextrem

**Pull request merged**